base/applications/charmap/map.c

```diff
@@ -111,6 +112,8 @@ FillGrid(PMAP infoPtr,
     for (y = 0; y < YCELLS; y++)
     for (x = 0; x < XCELLS; x++)
     {
+        if (i >= infoPtr->NumValidGlyphs) break;
```

Comment:
This break seems to stop just the inner for-loop. Is it what is intended there?

Reply:
Yes, ideally it would stop the entire loop but it will worst case run a single time on the next row, and will end at the correct column on the last line of valid glyphs.

Reply:
What would you think of me putting the `i >= infoPtr->NumValidGlyphs` in the for loop conditionals instead?  I'm on the fence about it ... but it would cause both for loops to terminate.

Reply:
The usual way to do that is to put a label after both loops and use `goto`